### PR TITLE
Pr80x l1t memory fixes

### PR DESCRIPTION
**Pull request by rekovic**
This PR contains 3 types of fixes concerning memory management by L1Trigger:
- fix for **memory leak** in Muon Barell Extrapolater which was introduced with merging of O2O
  (free allocated memory before every return).  This is already provided in 81x #16061 by @davidlange6 
- fix for **repetitive memory allocation** in Muon Overlap XML Parser to reset DOM vector pool and release all the associated memory back to the system before re-using read methods.
- fix for **Invalid writes** in Muon Barell ParamsHelper class by reserving the correct vector size.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
Some info on the change of memory usage by OMTF XMLConfigReader:

- Before fix: OMTF XMLConfigReader has no destructor (i.e. no deleting of parser), but delete DOM  documents at the end of read methods (multiple calls):
<pre>
  PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+  COMMAND
 7342 rekovic   20   0 2326m 1.4g 121m R 100.1  6.0   2:23.00 cmsRun                                                                                                                                
</pre> 
- After adding destructor and reseting parser's document pool in read methods:
<pre>
  PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+  COMMAND
 12196 rekovic   20   0 1922m 922m 121m R 97.5  3.9   2:12.69 cmsRun
</pre>

Reduction in actual physical memory consumption (in RES) is about 500m.

**Comment by rekovic** [comment]
+1

File: L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelParamsHelper.cc [hunk 1]
```diff
@@ -35,8 +35,8 @@ void L1TMuonBarrelParamsHelper::configFromPy(std::map<std::string, int>& allInts
 
 	setAssLUTPath(AssLUTpath);
 	///Read Pt assignment Luts
-	std::vector<LUT> pta_lut(0); pta_lut.reserve(19);
-	std::vector<int> pta_threshold(6); pta_threshold.reserve(10);
+	std::vector<LUT> pta_lut(19);
```
Comment (davidlange6) [inline L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelParamsHelper.cc]: @rekovic - careful - I believe that pta_lut should be reserve not resize - see that there is a push_back (but only 1 push_back.. )  on line 379 (this code is extremely confusing)

please confirm...

Comment (rekovic) [inline L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelParamsHelper.cc]: @davidlange6  Yes, you are right.  Pushing back on pta_lut vector will place beyond index 18, 
which is not good.  Changing back to reserve for pta_lut.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**